### PR TITLE
workflows: add self-hosted runners

### PR DESCRIPTION
**Pull request by jonchang**

### CODE REVIEW

File: .github/workflows/build-bottles.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+          git checkout -f FETCH_HEAD
+      - name: Cleanup
+        run: |
+          BREW_LIST=$(brew list 2>/dev/null)
+          if [ -n "$BREW_LIST" ]; then
+            echo ">> brew uninstall"
+            brew uninstall --force --ignore-dependencies $BREW_LIST
+          fi
+          rm -rf bottles
+          rm -rf *
```
Comment (sjackman) [inline .github/workflows/build-bottles.yml]: What's it cleaning up here? Does GitHub Actions not provide a clean space to each runner?

File: .github/workflows/build-bottles.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+          find . /usr/local/Homebrew -name .git/gc.log -delete
+          brew update-reset /usr/local/Homebrew
+          brew update
+      - name: Build bottles
+        run: |
+          mkdir bottles
+          cd bottles
+          brew test-bot --ci-pr
```
Comment (sjackman) [inline .github/workflows/build-bottles.yml]: `--ci-pr` should be autodetected from the environment variables?

File: .github/workflows/build-bottles.yml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    runs-on: ${{ matrix.version }}
+    env:
+      HOMEBREW_GIT_NAME: BrewTestBot
+      HOMEBREW_GIT_EMAIL: 'homebrew-test-bot@lists.sfconservancy.org'
+      PATH: '/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin'
+      HOMEBREW_DEVELOPER: 1
+      HOMEBREW_NO_ANALYTICS: 1
+      HOMEBREW_NO_AUTO_UPDATE: 1
+    steps:
+      - name: Setup tap
```
Comment (MikeMcQuaid) [inline .github/workflows/build-bottles.yml]: Could this use actions/checkout@master?

File: .github/workflows/build-bottles.yml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+          cd $(brew --repo ${{ github.repository }})
+          git config remote.origin.url https://github.com/Homebrew/homebrew-core/
+          git fetch origin refs/pull/${{ github.event.pull_request.number }}/merge
+          git checkout -f FETCH_HEAD
+      - name: Cleanup
+        run: |
+          BREW_LIST=$(brew list 2>/dev/null)
+          if [ -n "$BREW_LIST" ]; then
+            echo ">> brew uninstall"
+            brew uninstall --force --ignore-dependencies $BREW_LIST
```
Comment (MikeMcQuaid) [inline .github/workflows/build-bottles.yml]: `brew test-bot` should handle this already. If it doesn't: I suggest it's added.

File: .github/workflows/build-bottles.yml [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        with:
+          name: bottles
+          path: bottles
+      - name: Cleanup
+        if: always()
+        run: |
+          BREW_LIST=$(brew list 2>/dev/null)
+          if [ -n "$BREW_LIST" ]; then
+            echo ">> brew uninstall"
+            brew uninstall --force --ignore-dependencies $BREW_LIST
```
Comment (MikeMcQuaid) [inline .github/workflows/build-bottles.yml]: `brew test-bot` should definitely handle this already.

File: .github/workflows/build-bottles.yml (hunk 1 above)
Comment (MikeMcQuaid) [reply .github/workflows/build-bottles.yml]: I'd suggest this cleanup logic go into `brew test-bot`, too.

Comment (jonchang) [reply .github/workflows/build-bottles.yml]: > Does GitHub Actions not provide a clean space to each runner?

For GitHub hosted runners, yes, but this is basically like Jenkins where we need to do all of that.

File: .github/workflows/build-bottles.yml (hunk 2 above)
Comment (jonchang) [reply .github/workflows/build-bottles.yml]: This copies the Jenkins configuration.

File: .github/workflows/build-bottles.yml (hunk 3 above)
Comment (jonchang) [reply .github/workflows/build-bottles.yml]: This copies the Jenkins configuration. I believe test bot looks at the current brew repository, not its current directory.

File: .github/workflows/build-bottles.yml (hunk 4 above)
Comment (jonchang) [reply .github/workflows/build-bottles.yml]: It doesn’t. I need to figure out what environment variables test bot is looking at on Jenkins to determine its cleanup behavior.

File: .github/workflows/build-bottles.yml (hunk 3 above)
Comment (MikeMcQuaid) [reply .github/workflows/build-bottles.yml]: I'm not sure I understand this, can you elaborate a bit?

File: .github/workflows/build-bottles.yml (hunk 4 above)
Comment (MikeMcQuaid) [reply .github/workflows/build-bottles.yml]: Sounds good, thanks 👍

File: .github/workflows/build-bottles.yml (hunk 1 above)
Comment (MikeMcQuaid) [reply .github/workflows/build-bottles.yml]: Sounds good!

File: .github/workflows/build-bottles.yml (hunk 2 above)
Comment (MikeMcQuaid) [reply .github/workflows/build-bottles.yml]: Think it could be removed there too!

**Pull request merged**